910 guideline for roles (#946)
NoroffFEU/agency.noroff.dev: documentation/Team-roles-guidelines.md

   - Serve as a servant leader, putting the needs of the team first and helping them achieve their goals.

2. **Process Guardian:**
   - Ensure that the Scrum framework is understood and followed by the team.
   - Facilitate Scrum events, such as Sprint Planning, Daily Stand-ups, Sprint Review, and Sprint Retrospective.

3. **Coordinator:**
   - Promote effective communication within the team and acts as a link between teams. 
   - Address conflicts and foster collaboration among team members.
   - Promote collaboration between teams, Scrum masters meeting such as Sprint planning and spring retrospective.
   - Establish a dedicated channel for Scrum Masters to collaborate and communicate effectively.

4. **Continuous Improvement:**
   - Encourage the team to regularly reflect on their processes and identify areas for improvement.
   - Help the team adapt to change and continuously refine their working practices.

### Quality Assurance (QA):

1. **Responsability:**
   - Quality Assurance serves as the primary reviewer for the project, ensuring thorough assessment and validation of deliverables.

>    - Review the code changes and provide feedback.QA may verify that the reported issue has been adequately addressed and resolved. 

Reviewer: ```suggestion
   - Review the code changes and provide feedback. QA may verify that the reported issue has been adequately addressed and resolved. 
```

Author: made the suggested change.

>    - QA is responsible for merging, and if the issue is properly linked, it should automatically close itself. However, it sometimes encounters bugs.

Reviewer: ```suggestion
QA handles the merging process, and if the issue is correctly linked, it should close automatically. However, this process can occasionally experience bugs.
```

   - Ensure that quality is maintained throughout the development process.
   - Collaborate with developers to understand user stories and acceptance criteria.

>    - Establish a dedicated channel for Quality Assurances to collaborate and communicate effectively.

Reviewer: ```suggestion
   - Establish a dedicated channel for all Quality Assurance teams to collaborate and communicate effectively.
```

Author: maybe better: channel for QAs from all teams

2. **Test Planning and Execution:**
   - Create comprehensive test plans based on user stories and acceptance criteria.
   - Execute testing activities, including functional, regression, and performance testing.
   - Developers should support QA in creating test routines that can be used to benchmark the API throughout the project. 

3. **Automation:**
   - Identify opportunities for test automation to increase efficiency and reduce manual testing efforts.
   - Collaborate with developers to integrate automated testing into the CI/CD pipeline.

4. **Defect Management:**
   - Track and manage defects, working closely with developers to resolve issues.
   - Provide feedback to the development team on the quality of deliverables.

5. **Communication:**
   - Facilitating transparent and open communication with both Scrum Masters and developers to nurture a collaborative working environment.
   - Establish a dedicated channel for Quality Assurances to collaborate and communicate effectively.

### Developer:

1. **Coding and Implementation:**

>    - Write clean, maintainable, and efficient code based on requirements.

Reviewer: ```suggestion
   - Write clean, maintainable, and efficient code based on the requirements.
```

   - Collaborate with other developers to ensure consistency in coding practices.

2. **Collaboration:**
   - Engage actively in Scrum events, such as Sprint Planning and Daily Stand-ups.
   - Foster a collaborative environment, sharing knowledge and skills with team members.

3. **Adaptability:**
   - Embrace changes in requirements and be flexible in adapting to evolving project needs.
   - Continuously learn and apply new technologies and best practices.

4. **Code Review and Refactoring:**
   - Participate in code reviews to ensure code quality and adherence to coding standards.
   - Proactively identify opportunities for code refactoring to improve maintainability.

>    - Developers are required to verify that their issues have been merged and closed. Do NOT delete issues. 

Reviewer: ```suggestion
   - Developers are required to verify that their PRs have been merged and issues closed. Do NOT delete issues. A branch can safely be deleted when the PR is merged.
```
